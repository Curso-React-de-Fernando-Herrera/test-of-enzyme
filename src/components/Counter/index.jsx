import PropTypes from 'prop-types'
const { useState } = require("react")

const Counter = ({ value= 0 }) => {
  const [counter, setCounter] = useState(value)

  const handleLessCounter = () => setCounter(counter - 1)
  const handlePlusCounter = () => setCounter(counter + 1)
  const handleResetCounter = () => setCounter(value)

  return <section>
    <h1>Conter</h1>
    <h3>{counter}</h3>
    <article>
      <div>
        <button onClick={handleLessCounter}>-</button>
        <button onClick={handlePlusCounter}>+</button>
      </div>
      <button onClick={handleResetCounter}>Reset</button>
    </article>
  </section>
}

Counter.propTypes = {
  value: PropTypes.number
}

export default Counter
