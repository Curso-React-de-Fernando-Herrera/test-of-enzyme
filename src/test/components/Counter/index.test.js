import Counter from 'components/Counter'
import { shallow } from 'enzyme'

describe('<Counter /> component should be', () => {
  test('rendered', () => {
    const wrapper = shallow(<Counter />)

    expect(wrapper).toMatchSnapshot()
  })

  test('a initial value of counter', () => {
    const initialValue = 0

    const wrapper = shallow(<Counter />)
    const renderedValue = wrapper.find('h3').text()

    expect(renderedValue).toBe(String(initialValue))
    expect(renderedValue).not.toBe(String(initialValue + 1))
  })

  test('a custom value of counter', () => {
    const initialValue = 100
    const wrapper = shallow( <Counter value={initialValue} />)

    const returnedValue = wrapper.find('h3').text()

    expect(returnedValue).toBe(String(initialValue))
    expect(returnedValue).not.toBe(String(initialValue + 1))
  })

  test('increment counter', () => {
    const wrapper = shallow( <Counter /> )
    const initialCounter = wrapper.find('h3').text()

    wrapper.find('button').at(1).simulate('click')

    const returnedCounter = wrapper.find('h3').text()
    expect(returnedCounter).toBe(String(+initialCounter + 1))
  })

  test('decrement counter', () => {
    const wrapper = shallow( <Counter /> )
    const initialCounter = wrapper.find('h3').text()

    wrapper.find('button').at(0).simulate('click')

    const returnedCounter = wrapper.find('h3').text()
    expect(returnedCounter).toBe(String(+initialCounter - 1))
  })

  test('reset counter', () => {
    const wrapper = shallow( <Counter /> )
    wrapper.find('button').at(0).simulate('click')
    
    const decrementedCounter = wrapper.find('h3').text()
    
    wrapper.find('button').at(2).simulate('click')
    const resetedCounter = wrapper.find('h3').text()

    expect(resetedCounter).toBe(String(+decrementedCounter + 1))
  })
})