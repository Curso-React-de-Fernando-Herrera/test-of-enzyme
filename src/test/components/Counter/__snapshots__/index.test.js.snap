// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> component should be rendered 1`] = `
<section>
  <h1>
    Conter
  </h1>
  <h3>
    0
  </h3>
  <article>
    <div>
      <button
        onClick={[Function]}
      >
        -
      </button>
      <button
        onClick={[Function]}
      >
        +
      </button>
    </div>
    <button
      onClick={[Function]}
    >
      Reset
    </button>
  </article>
</section>
`;
